contrib/babelfishpg_common/src/varchar.c

```diff
@@ -1068,7 +1068,20 @@ varchar2date(PG_FUNCTION_ARGS)
 	 * Set input to default '1900-01-01' if empty string encountered
 	 */
 	if (isEmptyOrWhitespace(str))
-		date = initializeToDefaultDate();
+	{
+		/* Extract date from default timestamp (1900-01-01 00:00:00) */
+		struct pg_tm tt;
+		fsec_t fsec = 0;
+		Timestamp ts = initializeToDefaultDatetime();
+
+		if (timestamp2tm(ts, NULL, &tt, &fsec, NULL, NULL) != 0)
+			ereport(ERROR,
+					(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
+					 errmsg("timestamp out of range")));
+
+		/* Convert pg_tm to DateADT */
+		date = date2j(tt.tm_year, tt.tm_mon, tt.tm_mday) - POSTGRES_EPOCH_JDATE;
+	}
```

Comment:
We could have still used the initializeToDefaultDate in common extension, right?
And using the plugin_ptr we can expose it to TSQL Extension as well.